# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1413035047 -7200
#      Sat Oct 11 15:44:07 2014 +0200
# Branch gui-release
# Node ID 54943eb0ce3751b330ba5ad1b281a515c31e12d4
# Parent  9582fad687303d7b5a1aaf5e5d549a1517f4846f
reorder eol modes in the editor settings

* settings-dialog.ui: order the eol mode such that their indexes correspond
  to the values of the enum value in scintilla

* file-editor-tab.cc (detect_eol_mode): use settings or OS eol mode as default
  if no detection is possible (e.g. empty file);
  (new_file): use static cast from int to eol enum instead of array

* settings-dialog.cc (constructor): choose 2 (EolUnix) as default eol mode if
  qscintilla is not available

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -1364,27 +1364,44 @@ file_editor_tab::detect_eol_mode ()
                   i++;
                 }
               else
                 count_cr++;
             }
         }
     }
 
-  QsciScintilla::EolMode eol_mode = QsciScintilla::EolUnix;
-  int count_max = count_lf;
+  // get default from OS or from settings
+#if defined (Q_OS_WIN32)
+  int os_eol_mode = QsciScintilla::EolWindows;
+#elif defined (Q_OS_MAC)
+  int os_eol_mode = QsciScintilla::EolMac;
+#else
+  int os_eol_mode = QsciScintilla::EolUnix;
+#endif
+QSettings *settings = resource_manager::get_settings ();
+QsciScintilla::EolMode eol_mode = static_cast<QsciScintilla::EolMode> (
+      settings->value("editor/default_eol_mode",os_eol_mode).toInt ());
 
+  int count_max = 0;
+
+  if (count_crlf > count_max)
+    {
+      eol_mode = QsciScintilla::EolWindows;
+      count_max = count_crlf;
+    }
   if (count_cr > count_max)
     {
       eol_mode = QsciScintilla::EolMac;
       count_max = count_cr;
     }
-  if (count_crlf > count_max)
+  if (count_lf > count_max)
     {
-      eol_mode = QsciScintilla::EolWindows;
+      eol_mode = QsciScintilla::EolUnix;
+      count_max = count_lf;
     }
 
   return eol_mode;
 }
 
 void
 file_editor_tab::update_eol_indicator ()
 {
@@ -1406,28 +1423,26 @@ void
 file_editor_tab::new_file (const QString &commands)
 {
   update_window_title (false); // window title (no modification)
 
   QSettings *settings = resource_manager::get_settings ();
 
   // set the eol mode from the settings or depending on the OS if the entry is
   // missing in the settings
-  QsciScintilla::EolMode eol_modes[] =
-    {QsciScintilla::EolWindows, QsciScintilla::EolUnix, QsciScintilla::EolMac};
-
 #if defined (Q_OS_WIN32)
   int eol_mode = QsciScintilla::EolWindows;
 #elif defined (Q_OS_MAC)
   int eol_mode = QsciScintilla::EolMac;
 #else
   int eol_mode = QsciScintilla::EolUnix;
 #endif
   _edit_area->setEolMode (
-      eol_modes[settings->value("editor/default_eol_mode",eol_mode).toInt ()]);
+    static_cast<QsciScintilla::EolMode> (
+      settings->value("editor/default_eol_mode",eol_mode).toInt ()));
 
   update_eol_indicator ();
 
   _edit_area->setText (commands);
   _edit_area->setModified (false); // new file is not modified yet
 }
 
 void
diff --git a/libgui/src/settings-dialog.cc b/libgui/src/settings-dialog.cc
--- a/libgui/src/settings-dialog.cc
+++ b/libgui/src/settings-dialog.cc
@@ -191,17 +191,17 @@ settings_dialog::settings_dialog (QWidge
 #if defined (Q_OS_WIN32)
   int eol_mode = QsciScintilla::EolWindows;
 #elif defined (Q_OS_MAC)
   int eol_mode = QsciScintilla::EolMac;
 #else
   int eol_mode = QsciScintilla::EolUnix;
 #endif
 #else
-  int eol_mode = 1;
+  int eol_mode = 2;
 #endif
   ui->combo_eol_mode->setCurrentIndex (
     settings->value ("editor/default_eol_mode",eol_mode).toInt () );
   ui->editor_auto_ind_checkbox->setChecked (
     settings->value ("editor/auto_indent", true).toBool ());
   ui->editor_tab_ind_checkbox->setChecked (
     settings->value ("editor/tab_indents_line",false).toBool ());
   ui->editor_bs_unind_checkbox->setChecked (
diff --git a/libgui/src/settings-dialog.ui b/libgui/src/settings-dialog.ui
--- a/libgui/src/settings-dialog.ui
+++ b/libgui/src/settings-dialog.ui
@@ -47,18 +47,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_2">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>560</width>
+            <height>353</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_17">
            <item>
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>Interface</string>
              </property>
@@ -692,34 +692,34 @@
                    <property name="text">
                     <string>Default eol mode</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="combo_eol_mode">
                    <property name="sizeAdjustPolicy">
-                    <enum>QComboBox::AdjustToMinimumContentsLength</enum>
+                    <enum>QComboBox::AdjustToContents</enum>
                    </property>
                    <property name="minimumContentsLength">
                     <number>7</number>
                    </property>
                    <item>
                     <property name="text">
-                     <string>Windows</string>
+                     <string>Windows (CRLF)</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
-                     <string>Unix</string>
+                     <string>Mac (CR)</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
-                     <string>Mac</string>
+                     <string>Unix (LF)</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                  <item>
                   <spacer name="horizontalSpacer_17">
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
@@ -1275,18 +1275,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_4">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>488</width>
+            <height>262</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <layout class="QGridLayout" name="gridLayout_7">
                <property name="sizeHint" stdset="0">
@@ -1540,18 +1540,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_5">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>474</width>
+            <height>318</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_18">
            <item>
             <widget class="QGroupBox" name="groupBox_3">
              <property name="title">
               <string>Display</string>
              </property>
@@ -1704,18 +1704,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_6">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>203</width>
+            <height>134</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_19">
            <item>
             <layout class="QGridLayout" name="gridLayout_14">
              <property name="topMargin">
               <number>0</number>
              </property>
@@ -1779,18 +1779,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_8">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>368</width>
+            <height>244</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_25">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_26">
              <property name="sizeConstraint">
               <enum>QLayout::SetDefaultConstraint</enum>
              </property>
@@ -2042,18 +2042,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_7">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>529</width>
+            <height>204</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_20">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QCheckBox" name="checkbox_allow_web_connect">
                <property name="text">
